### PR TITLE
JBPM-7340: Added Swagger Info to our extension. Info field (actually …

### PR DESCRIPTION
**Pull request by DuncanDoyle**
…the title) is a required field in a Swagger definition.

### CODE REVIEW

**Comment by mswiderski** [comment]
I'd actually skip version completely as we do not version the API at all

File: kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java [hunk 1]
```diff
@@ -69,27 +70,34 @@ public void init(KieServerImpl kieServer, KieServerRegistry registry) {
...
+    	beanConfig.getSwagger().setInfo(getInfo());
+    	beanConfig.setScan(true);
+		
     	initialized = true;
     }
+    
+    private Info getInfo() {
+    	Info info = new Info();
+    	info.setTitle("KIE-Server API");
+    	info.setVersion("7.8.0");
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java]: can we remove the version completely?

**Comment by MarianMacik** [comment]
jenkins retest this

File: kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java (hunk 1 above)
Comment (DuncanDoyle) [reply kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java]: Not really .... it's a required field in the specification: https://swagger.io/specification/#infoObject

The whole point of adding the "Info" object was to be spec compliant ... so removing the version field would defy the purpose.

Also, I still think this should define the version of the API, not the version of KIE-Server. It should indicate versioning of the RESTful interface. So, if we patch stuff -> minor version, if we add stuff or make backwards compatible changes -> minor, and backwards incompatible changes -> major.

Would this be something we could infer from the JAX-RS annotations? For example during build time?

Comment (mswiderski) [reply kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java]: in that case I'd say we should use major version of the project/product as the version of the API - in this case version would be set to 7. wdyt?

Comment (MarianMacik) [reply kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java]: @DuncanDoyle We use revapi for notifications of API changes for example so we can infer it from there. But in general API versioning should be the same as project one I think. All minor releases should or even must be backward-compatible, majors don't need to. So if we stick to this, I agree with @mswiderski.

Comment (mswiderski) [reply kie-server-parent/kie-server-services/kie-server-services-swagger/src/main/java/org/kie/server/services/swagger/SwaggerKieServerExtension.java]: changed the version to be 7.0 - in general refers to major version of the project but uses minor to accommodate unexpected situation

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by MarianMacik** [comment]
Jenkins retest this

**Comment by mswiderski** [comment]
Jenkins retest this

**Comment by zyfAnd** [comment]
@mswiderski Could you provide method jBPM project how to call external Restful API service(eg: springboot Restful API )? thankls a lot

**Pull request merged**